skbio/metadata/_testing.py

```diff
@@ -904,3 +905,245 @@ def test_has_positional_metadata(self):
         obj = self._positional_metadata_constructor_(
             2, positional_metadata={'foo': [1, 2], 'bar': ['abc', 'def']})
         self.assertTrue(obj.has_positional_metadata())
+
+
+class IntervalMetadataMixinTests:
+    def _set_up(self):
+        self.upper_bound = 9
+        self.im = IntervalMetadata(self.upper_bound)
+        self.intvls = [
+            {'locations': [(0, 1), (2, 9)], 'metadata': {'gene': 'sagA'}},
+            # repeat it
+            {'locations': [(0, 1)], 'metadata': {'gene': ['a'],
+                                                 'product': 'foo'}}]
+
+    def test_constructor_invalid(self):
+        with self.assertRaisesRegex(TypeError,
+                                    'You must provide `IntervalMetadata` '
+                                    'object.'):
+            self._interval_metadata_constructor_(0, '')
+
+    def test_constructor_interval_metadata_len_mismatch(self):
+        for i in [0, 1, 3, 100]:
+            with self.assertRaisesRegex(ValueError, '\(9\).*\(%d\)' % i):
+                self._interval_metadata_constructor_(i, self.im)
+
+    def test_constructor_no_interval_metadata(self):
+        for i, im in [(0, None), (self.upper_bound, self.im)]:
+            obj = self._interval_metadata_constructor_(i, im)
+            self.assertFalse(obj.has_interval_metadata())
+            self.assertIsInstance(obj.interval_metadata, IntervalMetadata)
+
+    def test_constructor_makes_shallow_copy_of_interval_metadata(self):
```

Comment:
Can you add a test case similar to `test_constructor_handles_missing_positional_metadata_efficiently`?